test/wasm2c/hello.txt

```diff
@@ -62,9 +62,11 @@ wasm_rt_func_type_t wasm2c_test_get_func_type(uint32_t param_count, uint32_t res
 
 /* import: 'wasi_snapshot_preview1' 'fd_write' */
 u32 w2c_wasi__snapshot__preview1_fd_write(struct w2c_wasi__snapshot__preview1*, u32, u32, u32, u32);
+void wasm2c_tailcall_w2c_wasi__snapshot__preview1_fd_write(void **instance_ptr, void *tail_call_stack, wasm_rt_tailcallee_t *next);
```

Comment:
I think it would be tricky to conditionally emit these based on whether it's needed for a particular import -- we'd need to generate it if the import is ever called with `return_call` **or** if it's ever the target of a "ref.func" (including in an elem segment or global initializer) **or** if it's exported. We could add that tracking to the BinaryReaderIR but it would need to happen in a bunch of places. We're not really tracking anything specific to tail calls at that point; it's more like "can we prove that this import is never tail-called by anybody." :-/

If you want, what we could do is expose the Features to CWriter, and then this could be disabled globally behind `--enable-tail-call` (or, probably someday soon, `--disable-tail-call`). Would that work for you?

Reply:
> If you want, what we could do is expose the Features to CWriter, and then this could be disabled globally behind --enable-tail-call (or, probably someday soon, --disable-tail-call). Would that work for you?

Sure, this could be a good alternative for us!

Reply:
Great, just pushed.